fahrschein/src/main/java/org/zalando/fahrschein/domain/Metadata.java

```diff
@@ -13,23 +15,25 @@
     private final OffsetDateTime occurredAt;
     private final OffsetDateTime receivedAt;
     private final String flowId;
+    private final Map<String, String> spanCtx;
 
     @JsonCreator
     @Deprecated
-    private Metadata(@JsonProperty("event_type") String eventType, @JsonProperty("eid") String eid, @JsonProperty("occurred_at") String occurredAt, @JsonProperty("received_at") String receivedAt, @JsonProperty("flow_id") String flowId) {
-        this(eventType, eid, occurredAt == null ? null : OffsetDateTime.parse(occurredAt), receivedAt == null ? null : OffsetDateTime.parse(receivedAt), flowId);
+    private Metadata(@JsonProperty("event_type") String eventType, @JsonProperty("eid") String eid, @JsonProperty("occurred_at") String occurredAt, @JsonProperty("received_at") String receivedAt, @JsonProperty("flow_id") String flowId, @JsonProperty("span_ctx") Map<String, String> spanCtx) {
+        this(eventType, eid, occurredAt == null ? null : OffsetDateTime.parse(occurredAt), receivedAt == null ? null : OffsetDateTime.parse(receivedAt), flowId, spanCtx);
     }
 
-    public Metadata(String eventType, String eid, OffsetDateTime occurredAt, OffsetDateTime receivedAt, String flowId) {
+    public Metadata(String eventType, String eid, OffsetDateTime occurredAt, OffsetDateTime receivedAt, String flowId, Map<String, String> spanCtx) {
         this.eventType = eventType;
         this.eid = eid;
         this.occurredAt = occurredAt;
         this.receivedAt = receivedAt;
         this.flowId = flowId;
+        this.spanCtx = spanCtx;
```

Comment:
Should we create an immutable copy here? All other properties are immutable.

I'd also suggest adding another constructor without the span, to stay compatible with existing code.

Reply:
@jhorstmann I rolled back changes to constructors and use `JsonAnySetter` to catch span context.